api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -0,0 +1,200 @@
+  implicit val materializer: ActorMaterializer
+  val log: LoggingAdapter
+
+  implicit val timeout: Timeout
+
+  val splitLines = Framing.delimiter(ByteString("\n"), 2048, allowTruncation = true)
+
+  val institutionsPath =
+    path("institutions") {
+      val institutionsActor = system.actorSelection("/user/institutions")
+      get {
+        val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
+        onComplete(fInstitutions) {
+          case Success(institutions) =>
+            complete(ToResponseMarshallable(Institutions(institutions)))
+          case Failure(error) =>
+            log.error(error.getLocalizedMessage)
+            complete(HttpResponse(StatusCodes.InternalServerError))
+        }
+      }
+    }
+
+  val institutionByIdPath =
+    path("institutions" / Segment) { fid =>
+      extractExecutionContext { executor =>
+        val institutionsActor = system.actorSelection("/user/institutions")
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fInstitutionDetails = institutionDetails(fid, institutionsActor, filingsActor)
+          onComplete(fInstitutionDetails) {
+            case Success(institution) =>
+              filingsActor ! Shutdown
+              complete(ToResponseMarshallable(institution))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              log.error(error.getLocalizedMessage)
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val filingByPeriodPath =
+    path("institutions" / Segment / "filings" / Segment) { (fid, period) =>
+      extractExecutionContext { executor =>
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        val submissionActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fDetails: Future[FilingDetail] = filingDetailsByPeriod(period, filingsActor, submissionActor)
+          onComplete(fDetails) {
+            case Success(filingDetails) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(ToResponseMarshallable(filingDetails))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val submissionPath =
+    path("institutions" / Segment / "filings" / Segment / "submissions") { (fid, period) =>
+      val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+      post {
+        val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        submissionsActor ! CreateSubmission
+        val fLatest = (submissionsActor ? GetLatestSubmission).mapTo[Submission]
```

Comment:
This discussion veered off from the PR code review a bit. I think it's worthwhile to discuss this, but maybe as part of another ticket. This PR already has a lot of opinions and back and forth, and it needs some closure. @kgudel I'm not sure I follow your logic completely, but if you can create a test case that showcases this implementation failing according to the corner case you describe, that would be extremely useful. One thing to keep in mind is that we are assuming eventual consistency and "last write wins". Perhaps this answers your question.